### ISSUE TITLE
`AWS` - Failing while configuring vault (StatefulSet timeout)

### ISSUE THREAD

**Issue by alechp** [kubefirst/kubefirst#1788]
### Which version of kubefirst are you using?

kubefirst-cli golang utility version: v2.2.15

### Which cloud provider?

AWS

### Which installation type?

CLI

### Which distributed Git provider?

GitHub

## Error

```
Error: the StatefulSet was not created within the timeout period

If a detailed error message was available, please make the necessary corrections before retrying.
You can re-run the last command to try the operation again.
```
- Follow up to: https://github.com/kubefirst/kubefirst/issues/1773#issuecomment-1696441165

### Status
- Gitops repo is created; workflow is successful
> <img width="858" alt="Screenshot 2023-08-30 at 8 35 24 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/0b91e5e5-37ac-45b8-8b08-7b2a39f42b19">
- Metaphor repo is created; workflow is blocked on publish
<img width="1242" alt="Screenshot 2023-08-30 at 8 36 03 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/d4acb1a9-7281-4ffc-bd08-d5f52e866fbc">
- Everything up until `Configuring Vault` completed; this step still showing 0% after 10 minutes
<img width="1393" alt="Screenshot 2023-08-30 at 8 36 41 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/f90ab621-8e2c-4d15-9c59-5622967dc637">

### Logs
Whole log file is too long to paste here (and also not sure whether it's leaking sensitive info) but here's the end of the log before it errors out:
```
2023-08-30T20:19 INF cmd/aws/create.go:1187 > Setting argocd username and password credentials
2023-08-30T20:19 INF cmd/aws/create.go:1198 > argocd username and password credentials set successfully
2023-08-30T20:19 INF cmd/aws/create.go:1200 > Getting an argocd auth token
2023-08-30T20:19 INF pkg/argocd/auth.go:75 > using argocd url http://localhost:8080
2023-08-30T20:19 INF cmd/aws/create.go:1207 > argocd admin auth token set
2023-08-30T20:19 INF cmd/aws/create.go:1238 > applying the registry application to argocd
2023-08-30T20:19 INF cmd/aws/create.go:1257 > waiting for vault pods to be ready 
2023-08-30T20:19 INF pkg/k8s/exec.go:281 > waiting for vault StatefulSet to be created using label app.kubernetes.io/instance=vault
2023-08-30T20:29 ERR pkg/k8s/exec.go:303 > the StatefulSet was not created within the timeout period
2023-08-30T20:29 ERR cmd/aws/create.go:1267 > Error finding Vault StatefulSet: the StatefulSet was not created within the timeout period
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by alechp** [comment]
- Relevant comment from Clay for someone else's issue
> ![image](https://github.com/kubefirst/kubefirst/assets/3058939/7b35c72c-53c2-4bb2-839e-14d01d621cb8)
- Confirmation that I was able to get AWS running before with the same stack, but haven't been able to replicate since
> ![image](https://github.com/kubefirst/kubefirst/assets/3058939/2b116b42-bc61-4be2-9887-b2e9c510837d)
- Noticing Principal IAM user error when I check the console manually. This could be related. However, it's odd to me that it passed both EKS deployment and Argo configuration only to fail at Vault config
> <img width="521" alt="Screenshot 2023-08-31 at 2 53 53 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/90821457-2a66-44eb-89b6-0f7f2f7e2f17">

**Comment by alechp** [comment]
- Noticed that external DNS was throwing an error
```
Failed to load logs: container "external-dns" in pod "external-dns-7ff5444dd5-95qv8" is waiting to start: CreateContainerConfigError  

Reason: BadRequest (400)
```
> <img width="2672" alt="Screenshot 2023-08-31 at 3 32 04 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/b646ff6e-2837-4fce-96dd-60c99cc02366">

- After restarting, found a new bug
`Error: secret "aws-auth" not found`
![image](https://github.com/kubefirst/kubefirst/assets/3058939/9a3f7428-7fcd-4b27-9417-7895f438bc1a)

- When I check Secrets, I see there's an `aws-secret`. Wondering whether:
a. This is a rename issue (ie. it's looking for `aws-auth` but should be looking for `aws-secret`
b. There's a second secret name `aws-auth` which isn't being generated

> <img width="513" alt="Screenshot 2023-08-31 at 3 34 06 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/23375fe0-2af8-4389-9e3a-f61a19467e83">

**Comment by alechp** [comment]
Looking for `aws-auth`:
> <img width="1086" alt="Screenshot 2023-08-31 at 4 20 38 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/b1e7b203-d19e-422b-bf04-e9b144ada0c5">

The only place I have been able to find reference to `aws-auth` is in an archived repo:
- https://github.com/kubefirst/gitops-gh-template/blob/364ca3b6950b549c67eb470c19088bdfa43245ed/registry/base/aws-auth.yaml#L4
- https://github.com/kubefirst/gitops-gh-template/blob/364ca3b6950b549c67eb470c19088bdfa43245ed/terraform/base/eks/variables.tf#L2

I was able to find `aws-secret` (which is created and available on the cluster) in the `ci-template` repo here:
- https://github.com/kubefirst/ci-template/blob/9125cbafabb658f3b5df4e9fe8ee4a42586cf73e/terraform/secret/main.tf#L17
> <img width="656" alt="Screenshot 2023-08-31 at 4 23 20 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/4e0c017f-8642-46ab-a0a4-472dcf8d7cf6">

**Comment by alechp** [comment]
Found a small bug on https://docs.kubefirst.io/aws/explore/vault
> <img width="1364" alt="Screenshot 2023-08-31 at 4 44 46 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/9da854bf-0cd8-48eb-9728-ddb9d6dadf96">

- Command listed: 
> `~/.k1/tools/kubectl -n vault get secrets/vault-unseal-secret --template='{{index .data "root-token"}}' | base64 -d`

- Actual path to`/tools/`: 
> <img width="414" alt="Screenshot 2023-08-31 at 4 46 25 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/8682771d-f10d-45b5-8d5f-a6b8a808244e">

Options:
1. Install tools in root directory like the docs mention (easier, especially since multiple clusters would use common binary)
2. Update docs to mention project name (eg. `~/.k1/<MY_CLUSTER>/tools/kubectl`)

**Comment by Claywd** [comment]
@alechp all this was great work. The ci-template I believe is no longer used. We are using the gitops-template now. I believe this is fixed in #1789. Feel free to re-open if not.

**Comment by alechp** [comment]
Deploy finished!! Thank you @Claywd. Definitely fixed the `aws-auth` issue above

<img width="1012" alt="Screenshot 2023-08-31 at 7 45 24 PM" src="https://github.com/kubefirst/kubefirst/assets/3058939/5d86cf24-c729-4fd2-87b4-2cdd9c95a7c3">
